cpp/src/io/json/json_column.cu

```diff
@@ -0,0 +1,846 @@
+      }
+
+      // For string columns return ["offsets", "char"] schema
+      if (target_type.id() == type_id::STRING) {
+        return {std::move(col), {{"offsets"}, {"chars"}}};
+      }
+      // Non-string leaf-columns (e.g., numeric) do not have child columns in the schema
+      else {
+        return {std::move(col), {}};
+      }
+      break;
+    }
+    case json_col_t::StructColumn: {
+      std::vector<std::unique_ptr<column>> child_columns;
+      std::vector<column_name_info> column_names{};
+      size_type num_rows{json_col.num_rows};
+      // Create children columns
+      for (auto const& col_name : json_col.column_order) {
+        auto const& col = json_col.child_columns.find(col_name);
+        column_names.emplace_back(col->first);
+        auto& child_col            = col->second;
+        auto [child_column, names] = json_column_to_cudf_column2(
+          child_col, d_input, options, get_child_schema(col_name), stream, mr);
+        CUDF_EXPECTS(num_rows == child_column->size(),
+                     "All children columns must have the same size");
+        child_columns.push_back(std::move(child_column));
+        column_names.back().children = names;
+      }
+      auto [result_bitmask, null_count] = make_validity(json_col);
+      return {
+        make_structs_column(
+          num_rows, std::move(child_columns), null_count, std::move(result_bitmask), stream, mr),
+        column_names};
+      break;
+    }
+    case json_col_t::ListColumn: {
+      size_type num_rows = json_col.child_offsets.size() - 1;
+      std::vector<column_name_info> column_names{};
+      column_names.emplace_back("offsets");
+      column_names.emplace_back(json_col.child_columns.begin()->first);
+
+      // Note: json_col modified here, reuse the memory
+      auto offsets_column = std::make_unique<column>(
+        data_type{type_id::INT32}, num_rows + 1, json_col.child_offsets.release());
+      // Create children column
+      auto [child_column, names] =
+        json_column_to_cudf_column2(json_col.child_columns.begin()->second,
+                                    d_input,
+                                    options,
+                                    get_child_schema(json_col.child_columns.begin()->first),
+                                    stream,
+                                    mr);
+      column_names.back().children      = names;
+      auto [result_bitmask, null_count] = make_validity(json_col);
+      return {make_lists_column(num_rows,
+                                std::move(offsets_column),
+                                std::move(child_column),
+                                null_count,
+                                std::move(result_bitmask),
+                                stream,
+                                mr),
+              std::move(column_names)};
+      break;
+    }
+    default: CUDF_FAIL("Unsupported column type, yet to be implemented"); break;
+  }
+
+  return {};
```

Comment:
This looks unreachable, can we remove this?

Reply:
yes.